evm_arithmetization/src/generation/jumpdest.rs

```diff
@@ -0,0 +1,191 @@
+use std::cmp::max;
+use std::{
+    collections::{BTreeSet, HashMap},
+    fmt::Display,
+    ops::{Deref, DerefMut},
+};
+
+use itertools::{sorted, Itertools};
+use keccak_hash::H256;
+use serde::{Deserialize, Serialize};
+
+/// Each `CodeAddress` can be called one or more times, each time creating a new
+/// `Context`. Each `Context` will contain one or more offsets of `JUMPDEST`.
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub struct ContextJumpDests(pub HashMap<usize, BTreeSet<usize>>);
+
+/// The result after proving a `JumpDestTableWitness`.
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub(crate) struct JumpDestTableProcessed(HashMap<usize, Vec<usize>>);
+
+/// Map `CodeAddress -> (Context -> [JumpDests])`
+#[derive(PartialEq, Eq, Debug, Clone, Serialize, Deserialize, Default)]
+pub struct JumpDestTableWitness(HashMap<H256, ContextJumpDests>);
```

Comment:
I had already asked but I think my comment got resolved, why do we need actual wrappers here instead of type aliases? Most of the implementation below would come for free instead of having to reimplement everything. Or if it's more convenient for merging tables we could just expose the inner map like what `ContextJumpDests` is doing?

Reply:
here was my reponse:

>When I was diving into this, it made it more clear what the purpose of both types was. But otherwise it provides a bit of encapsulation and type safety.

I have a mild preference for this New Type pattern over type aliases, but it is not the hill I am going to die on.

Reply:
Ok, let's keep it.
On another note, why do we need to expose the inner map in `ContextJumpDests`? Also it is public but only used in this crate I think?